lib/block-editor.php

```diff
@@ -86,7 +86,7 @@ function gutenberg_get_block_categories( $editor_name_or_post ) {
 	 *
 	 * @param array[] $default_categories Array of categories for block types.
 	 */
-	$block_categories = apply_filters( "block_categories_{$editor_name}", $default_categories );
+	$block_categories = apply_filters( 'block_categories_all', $default_categories );
```

Comment:
When I read https://github.com/WordPress/wordpress-develop/pull/1118#issuecomment-823507573 and this issue's description I thought that we'd do this:  check if `block_categories_${editor_name}` exists, and don't run the deprecated `block_categories` filter if it does (same for the other filters). Would it work if we limit this PR to do that and so fixing the warnings when `WP_DEBUG` is `true`?

Related: what do you think of not introducing a `*_all` filters at the moment? I think it'd be fine to do in the case all take the same config:

```
add_filter( 'block_categories_post_editor', callback );
add_filter( 'block_categories_site_editor', callback );
...
```

I'm thinking that `*_all` is just a particular use case (all editor use the same settings) but there may be more (post and site use the same but widgets doesn't, etc), that's why I'm not fully convinced is worth introducing the `_all` filters at the moment.

Reply:
>  check if `block_categories_${editor_name}` exists

What does this mean? That there are subscribers (`has_filter`), or that there are consumers? I'm trying to understand how this would help.

Reply:
> Related: what do you think of not introducing a *_all filters at the moment? I think it'd be fine to do in the case all take the same config:

It's a completely opposite direction that @youknowriad advocates for in https://github.com/WordPress/gutenberg/pull/31027#discussion_r617464608 and expands on in https://github.com/WordPress/gutenberg/pull/31027#issuecomment-824011333.